src/ports/picks/component.ts

```diff
@@ -37,19 +37,26 @@ export function createPicksComponent(components: Pick<AppComponents, 'pg'>): IPi
   }
 
   async function getPicksByItemId(itemId: string, options: GetPicksByItemIdParameters): Promise<DBGetFilteredPicksWithCount[]> {
-    const { limit, offset } = options
-    const result = await pg.query<DBGetFilteredPicksWithCount>(SQL`
-        SELECT user_address, COUNT(*) OVER() as picks_count
-        FROM (
-          SELECT DISTINCT favorites.picks.user_address FROM favorites.picks, favorites.voting
-          WHERE favorites.picks.item_id = ${itemId}
-          AND favorites.voting.user_address = favorites.picks.user_address AND favorites.voting.power >= ${
-            options.power ?? DEFAULT_VOTING_POWER
-          }
-        ) AS temp
-        ORDER BY user_address
-        LIMIT ${limit} OFFSET ${offset}
-    `)
+    const { limit, offset, power, userAddress } = options
+    const query = SQL`SELECT user_address, COUNT(*) OVER() as picks_count`
+
+    if (userAddress) {
+      query.append(SQL`, (CASE WHEN user_address = ${userAddress} THEN 1 ELSE 0 END)::BOOLEAN AS picked_by_user`)
```

Comment:
What do you think about omitting the boolean here as the number will work as well for the order?
```suggestion
      query.append(SQL`, (CASE WHEN user_address = ${userAddress} THEN 1 ELSE 0 END) AS picked_by_user`)
```